benchmarks/src/bin/tpch.rs

```diff
@@ -285,11 +285,12 @@ async fn execute_query(
     if debug {
         println!("=== Logical plan ===\n{:?}\n", plan);
     }
-    let plan = ctx.optimize(plan)?;
     if debug {
-        println!("=== Optimized logical plan ===\n{:?}\n", plan);
+        let optimized_plan = ctx.optimize(plan)?;
+        println!("=== Optimized logical plan ===\n{:?}\n", optimized_plan);
     }
-    let physical_plan = ctx.create_physical_plan(&plan).await?;
+    // note that `create_physical_plan` will optimize the plan so we pass the unoptimized plan
```

Comment:
Is there error or different result for your plan, If it is optimized by some rules more times?
I also this in my preview pr.

Reply:
For example:
A+B, we will get the coercion type C. After the first optimization, will get `cast(A AS C) + CAST(B AS C)`.
After the second optimization, we may get the coercion type D from `cast(A AS C) + CAST(B AS C)`.

Reply:
The original code was running the optimizer twice, which was not necessary. With the new rule there was a problem optimizing twice. I will look at this again today and write up an issue or make it safe to optimize twice.

Reply:
I reverted this change